src/Interpreters/Access/InterpreterGrantQuery.cpp

```diff
@@ -366,6 +366,13 @@ BlockIO InterpreterGrantQuery::execute()
     AccessRightsElements elements_to_grant, elements_to_revoke;
     collectAccessRightsElementsToGrantOrRevoke(query, elements_to_grant, elements_to_revoke);
 
+    if (query.current_grants)
+    {
+        AccessRights new_rights(elements_to_grant);
+        new_rights.makeIntersection(*current_user_access->getAccessRightsWithImplicit());
```

Comment:
I think it should not be "WithImplicit" just because it doesn't feel right to convert implicit grants to explicit ones.